### PR TITLE
Get Children Blocks 

### PR DESCRIPTION
**Pull request by bspradling**
## Description 

Generic implementation of retrieving block information. I have yet to enumerate different fields for specific Block types. 

Fixed some visibility issues for people using this client externally. 
Fixed an casing issue when deserializing Url types

Recreating PR to be a branch instead of fork.

### CODE REVIEW

File: src/models.rs [hunk 1]
```diff
@@ -136,12 +136,164 @@ pub struct Page {
...
+        #[serde(flatten)]
+        common: BlockCommon,
+        toggle: TextAndChildren,
+    },
+    ChildPage {
+        #[serde(flatten)]
+        common: BlockCommon,
+        child_page: ChildPageFields,
+    },
+    Unsupported {},
```
Comment (jakeswenson) [inline src/models.rs:236]: not sure if this is a good usecase of https://serde.rs/variant-attrs.html#other

File: src/models.rs [hunk 2]
```diff
@@ -136,12 +136,164 @@ pub struct Page {
 }
 
 #[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone)]
-pub struct Block {}
+#[serde(rename_all = "snake_case")]
+pub enum BlockType {
```
Comment (jakeswenson) [inline src/models.rs:140]: not needed

File: src/models.rs [hunk 3]
```diff
@@ -136,12 +136,164 @@ pub struct Page {
...
+            Block::NumberedListItem {
+                common,
+                numbered_list_item: _,
+            } => &common.id,
+            Block::ToDo { common, to_do: _ } => &common.id,
+            Block::Toggle { common, toggle: _ } => &common.id,
+            Block::ChildPage {
+                common,
+                child_page: _,
+            } => &common.id,
```
Comment (jakeswenson) [inline src/models.rs:273]: What about this:
```suggestion
        use Self::*;
        match self {
            Paragraph { common, .. }
            | Heading1 { common, .. }
            | Heading2 { common, ..}
            | Heading3 { common, .. }
            | BulletedListItem { common, .. }
            | NumberedListItem { common, .. }
            | ToDo { common, .. }
            | Toggle { common, .. }
            | ChildPage { common, .. } => &common.id,
```

**Review by jakeswenson — APPROVED** [review]

File: src/models.rs (hunk 1 above)
Comment (bspradling) [reply src/models.rs]: I think it makes sense as a catch all

**Pull request merged**